Model/Api/CreateOrder.php

```diff
@@ -192,21 +197,24 @@ public function execute(
 
             $this->preProcessWebhook($immutableQuote->getStoreId());
 
-            $payload = $this->request->getContent();
             $transaction = json_decode($payload);
 
             /** @var Quote $quote */
             $quote = $this->orderHelper->prepareQuote($immutableQuote, $transaction);
-            $this->validateQuoteData($quote, $transaction);
 
-            /** @var \Magento\Sales\Model\Order $createOrderData */
-            $createOrderData = $this->orderHelper->preAuthCreateOrder($quote, $transaction);
+            /** @var \Magento\Sales\Model\Order $createdOrder */
+            $createdOrder = $this->orderHelper->processExistingOrder($quote, $transaction);
+
+            if (! $createdOrder) {
+                $this->validateQuoteData($quote, $transaction);
```

Comment:
The validation was moved after the check for an existing order because it can fail on inventory validation if the last item(s) of the product got reserved in that order. In processExistingOrder we call validation of totals.